src/installer/records.py

```diff
@@ -49,6 +61,17 @@ def __repr__(self):
             self.path, self.hash_, self.size,
         )
 
+    def validate(self, data):
+        # type: (bytes) -> bool
+        if self.size is not None and len(data) != self.size:
+            return False
+
+        if self.hash_:
+            hash = hashlib.new(self.hash_.name, data)
+            return hash.digest() == self.hash_.digest
```

Comment:
I personally would favour just compare the base64-encoded values, i.e. instead of decoding the `Hash.value` back, encode the incoming has instead:

```python
digest = hashlib.new(self.hash_.name, data).digest()
value = base64.urlsafe_b64encode(digest).decode("ascii").rstrip("=")
return self.hash_.value == value
```

The difference is minimal, but I like to stick to the security best practice of always comparing the encoded result.